docs/manage-fileset-metadata-using-gravitino.md

```diff
@@ -0,0 +1,422 @@
+
+| Supported modification             | JSON                                                         | Java                                          |
+|------------------------------------|--------------------------------------------------------------|-----------------------------------------------|
+| Rename fileset                     | `{"@type":"rename","newName":"fileset_renamed"}`             | `FilesetChange.rename("fileset_renamed")`     |
+| Update comment                     | `{"@type":"updateComment","newComment":"new_comment"}`       | `FilesetChange.updateComment("new_comment")`  |
+| Set a fileset property             | `{"@type":"setProperty","property":"key1","value":"value1"}` | `FilesetChange.setProperty("key1", "value1")` |
+| Remove a fileset property          | `{"@type":"removeProperty","property":"key1"}`               | `FilesetChange.removeProperty("key1")`        |
+
+### Drop a fileset
+
+You can remove a fileset by sending a `DELETE` request to the `/api/metalakes/{metalake_name}
+/catalogs/{catalog_name}/schemas/{schema_name}/filesets/{fileset_name}` endpoint or just use the
+Gravitino Java client. The following is an example of dropping a fileset:
+
+<Tabs>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X DELETE -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" \
+http://localhost:8090/api/metalakes/metalake/catalogs/catalog/schemas/schema/filesets/fileset
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+// ...
+// Assuming you have just created a Fileset catalog named `catalog`
+Catalog catalog = gravitinoMetaLake.loadCatalog(NameIdentifier.of("metalake", "catalog"));
+
+FilesetCatalog filesetCatalog = catalog.asFilesetCatalog();
+
+// Drop a fileset
+filesetCatalog.dropFileset(NameIdentifier.of("metalake", "catalog", "schema", "fileset"));
+// ...
+```
+
+</TabItem>
+</Tabs>
+
+For `MANAGED` fileset, the physical location of the fileset will be deleted when this fileset is
+dropped. For `EXTERNAL` fileset, only the metadata of the fileset will be removed.
+
+### List filesets
+
+You can list all filesets in a schema by sending a `GET` request to the `/api/metalakes/
+{metalake_name}/catalogs/{catalog_name}/schemas/{schema_name}/filesets` endpoint or just use the
+Gravitino Java client. The following is an example of list all the filesets in a schema:
+
+<Tabs>
+<TabItem value="shell" label="Shell">
+
+```shell
+curl -X GET -H "Accept: application/vnd.gravitino.v1+json" \
+-H "Content-Type: application/json" \
+http://localhost:8090/api/metalakes/metalake/catalogs/catalog/schemas/schema/filesets
+```
+
+</TabItem>
+<TabItem value="java" label="Java">
+
+```java
+// ...
+Catalog catalog = gravitinoMetaLake.loadCatalog(NameIdentifier.of("metalake", "catalog"));
+
+FilesetCatalog filesetCatalog = catalog.asFilesetCatalog();
+NameIdentifier[] identifiers =
+    filesetCatalog.listFilesets(Namespace.ofFileset("metalake", "catalog", "schema"));
```

Comment:
I think there's another issue. The `metalake` in Namespace seems redundant. The new GravitinoClient, we have declared the name of the current metalake. It is not related to this MR. Never mind.

Reply:
Yeah, it is unrelated, and we are working on the client refactoring things.